.github/workflows/bump_dev_version_nightly.yml

```diff
@@ -0,0 +1,116 @@
+
+
+          - id: should_run
+            continue-on-error: true
+            name: check latest commit is less than a day
+            if: ${{ github.event_name == 'schedule' }}
+            run: test -z $(git rev-list  --after="24 hours" $(git log -n 1 --pretty=format:"%H")) && echo "::set-output name=should_run::false"
+            # If the latest commit is less than 24 hours old, the command git rev-list --after="24 hours" $(git log -n 1 --pretty=format:"%H") will return a non-empty string (list of commit ids in the last 24 hours), which will cause the test -z command to return false and the output should_run will not be set to false.
+            # test -z $STR checks if $STR is empty or not. If it is empty, it returns true, else false.
+
+  bump-version_tudat:
+    needs: check_recent_24h_commit
+    if: ${{ needs.check_recent_24h_commit.outputs.should_run != 'false' }}
+    runs-on: ubuntu-latest
+    name: Bump version in tudat repository
+    outputs:
+      new_version: ${{ steps.bump_version.outputs.new_version }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: develop
+
+      - name: Install bump2version
+        run: pip install bump2version
+
+      - name: Bump version
+        id: bump_version
+        run: |
+          git config --global user.email "actions@github.com"
+          git config --global user.name "GitHub Actions"
+          bump2version dev --config-file .bumpversion.cfg --verbose
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Push tag to tudat repository
+        run: git push origin --follow-tags
+        env:
+            GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+
+  bump_version_tudat_feedstock:
+    needs: check_recent_24h_commit
+    if: ${{ needs.check_recent_24h_commit.outputs.should_run != 'false' }}
+    runs-on: ubuntu-latest
+    name: bump version and create tag in tudat-feedstock
+    steps:
+        - name: checkout tudat-feedstock
+          uses: actions/checkout@v4
+          with:
+            repository: tudat-team/tudat-feedstock
+            token: ${{ secrets.<add_secret_name> }}
+            ref: develop
+
+        - name: Install bump2version
+          run: pip install bump2version
+
+        - name: Bump version on feedstock
+          run: |
+              git config --global user.email "actions@github.com"
+              git config --global user.name "GitHub Actions"
+              bump2version dev --config-file .bumpversion.cfg --verbose
+
+        - name: Reset build number in meta.yml to 0
+          shell: bash
+          run: |
+                sed -i 's/build = ".*" %}/build = "0" %}/' recipe/meta.yaml
+                git add recipe/meta.yaml
+                git commit --amend --no-edit
+
+        - name: Push tag to tudat-feedstock
+          run: git push origin --follow-tags
+          env:
+              GITHUB_TOKEN: ${{ secrets.<add_secret_name>}}
```

Comment:
@DominicDirkx replace <add_secret_name> with your personal access token. See the PR notes on how to create one.